pool/paymentmgr.go

```diff
@@ -595,71 +644,237 @@ func (pm *PaymentMgr) maturePendingPayments(height uint32) (map[uint32][]*Paymen
+
+	// Deduct the transaction fees from the pool fees being paid out.
+	fees := outputs[feeAddr.String()]
+	fees = fees - estFee
+
+	// Deduct any percentage rounding value differences from the pool
+	// fees being paid out.
+	if diff != zeroAmount {
+		fees = fees - diff
+	}
+
+	outputs[feeAddr.String()] = fees
+
+	// Generate the output set with decoded addresses.
+	outs := make(map[dcrutil.Address]dcrutil.Amount, len(outputs))
+	for sAddr, amt := range outputs {
+		addr, err := dcrutil.DecodeAddress(sAddr, pm.cfg.ActiveNet)
+		if err != nil {
+			return fmt.Errorf("unable to decode address: %v", err)
+		}
+
+		outs[addr] = amt
+	}
+
+	tx, err := txCreator.CreateRawTransaction(inputs, outs, nil, nil)
+	if err != nil {
+		return fmt.Errorf("unable to create raw transaction: %v", err)
+	}
+
+	txB, err := tx.Bytes()
 	if err != nil {
 		return err
 	}
-	for _, bundle := range eligiblePmts {
-		bundle.UpdateAsPaid(pm.cfg.DB, height, txid)
-		err = bundle.ArchivePayments(pm.cfg.DB)
+
+	for {
+		// Ensure the wallet is synced to the current chain height before
+		// proceeding with payments.
+		walletHeight, err := pm.cfg.WalletBestHeight()
 		if err != nil {
-			return err
+			return fmt.Errorf("unable to fetch best wallet height: %v", err)
+		}
+
+		if height == walletHeight {
+			break
 		}
 	}
-	err = pm.cfg.DB.Update(func(tx *bolt.Tx) error {
-		err = pm.persistTxFeeReserve(tx)
-		if err != nil {
-			return err
+
+	txBroadcaster := pm.cfg.FetchTxBroadcaster()
+	if txBroadcaster == nil {
+		return fmt.Errorf("tx broadcastor unset")
+	}
+
+	balanceReq := &walletrpc.BalanceRequest{
+		AccountNumber:         pm.cfg.WalletAccount,
+		RequiredConfirmations: int32(pm.cfg.ActiveNet.CoinbaseMaturity + 1),
+	}
+	balanceResp, err := txBroadcaster.Balance(context.TODO(), balanceReq)
+	if err != nil {
+		return fmt.Errorf("unable to get account balance: %v", err)
+	}
+
+	spendable := dcrutil.Amount(balanceResp.Spendable)
+
+	if spendable < tOut {
+		return fmt.Errorf("insufficient account funds, pool account "+
+			"has only %v to spend, however outgoing transaction will "+
+			"require spending %v", spendable, tOut)
+	}
+
+	// Sign the transaction.
+	signTxReq := &walletrpc.SignTransactionRequest{
+		SerializedTransaction: txB,
+		Passphrase:            []byte(pm.cfg.WalletPass),
+	}
+	signedTxResp, err := txBroadcaster.SignTransaction(context.TODO(),
+		signTxReq)
+	if err != nil {
+		return fmt.Errorf("unable to sign transaction: %v", err)
+	}
+
+	// Publish the transaction.
+	pubTxReq := &walletrpc.PublishTransactionRequest{
+		SignedTransaction: signedTxResp.Transaction,
+	}
+	pubTxResp, err := txBroadcaster.PublishTransaction(context.TODO(),
+		pubTxReq)
+	if err != nil {
+		return fmt.Errorf("unable to publish transaction: %v", err)
+	}
+
+	txid, err := chainhash.NewHash(pubTxResp.TransactionHash)
+	if err != nil {
+		return err
+	}
+
+	log.Tracef("paid a total of %v in tx %s, including %v in pool fees.",
```

Comment:
I would promote this from Trace to Debug, possibly even Info. It's really useful information and wont be logged too frequently.

Reply:
resolved in upcoming commit.